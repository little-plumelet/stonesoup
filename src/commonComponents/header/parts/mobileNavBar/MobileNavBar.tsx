import { MenuOutlined } from '@ant-design/icons';
import { Button, Drawer } from 'antd';
import { useState } from 'react';
import { NavBar } from '../navBar';

export function MobileNavBar() {
  const [open, setOpen] = useState(false);

  const showDrawer = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <>
      <Button
        data-testid="burger-menu"
        onClick={showDrawer}
        icon={<MenuOutlined />}
        shape="circle"
      />
      <Drawer placement="top" open={open} onClose={handleClose}>
        <NavBar
          navStyle={{
            display: 'flex',
            flexDirection: 'column',
            gap: '1rem',
          }}
        />
      </Drawer>
    </>
  );
}
