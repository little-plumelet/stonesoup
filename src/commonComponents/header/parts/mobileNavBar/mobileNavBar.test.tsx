import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { MobileNavBar } from './MobileNavBar';

describe('navbar', () => {
  it('should be rendered', () => {
    const component = render(
      <MemoryRouter>
        <MobileNavBar />
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });
  it('links work', () => {
    const history = createMemoryHistory();

    render(
      <Router location={history.location} navigator={history}>
        <MobileNavBar />
      </Router>
    );

    fireEvent.click(screen.getByTestId('burger-menu'));
    fireEvent.click(screen.getByText('Home'));
    expect(history.location.pathname).toBe('/');

    fireEvent.click(screen.getByTestId('burger-menu'));
    fireEvent.click(screen.getByText('About Us'));
    expect(history.location.pathname).toBe('/about');
  });
});
