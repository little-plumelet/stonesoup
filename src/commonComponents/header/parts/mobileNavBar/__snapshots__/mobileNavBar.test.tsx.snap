// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar should be rendered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-circle ant-btn-default ant-btn-icon-only"
        data-testid="burger-menu"
        type="button"
      >
        <span
          aria-label="menu"
          class="anticon anticon-menu"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="menu"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-circle ant-btn-default ant-btn-icon-only"
      data-testid="burger-menu"
      type="button"
    >
      <span
        aria-label="menu"
        class="anticon anticon-menu"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="menu"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
